augur/filter.py

```diff
@@ -234,7 +236,7 @@ def run(args):
     # filter by date
     num_excluded_by_date = 0
     if (args.min_date or args.max_date) and 'date' in meta_columns:
-        dates = get_numerical_dates(meta_dict, fmt="%Y-%m-%d")
+        dates = get_numerical_dates(meta_dict, fmt="%Y-%m-%d", exclude_ambiguous_dates=args.exclude_ambiguous_dates)
```

Comment:
This is a really clever solution to the problem! While this definitely implements the ambiguous date filter, it puts more information about augur filter into the `get_numerical_dates` function than it needs (e.g., tracking number of filtered records). This solution also limits ambiguous date filtering to cases when users also provide a min or max date. There will be many times when users want to filter all ambiguous dates regardless of a min/max date.

As an alternate approach, I would recommend the following changes:

  1. add a new filter block just above line 236 (`# filter by date`) that defines a new variable to track how many records were filtered by ambiguous date
  2. define a simple boolean function `is_date_ambiguous` that can test each entry of the metadata date column (see the inline comment about this below for an example function).
  3. loop through `meta_dict` dates for samples in `seq_keep`, keeping only those that pass the boolean function test
  4. add a print statement to the report at the end of the `run` function that reports how many samples were dropped due to ambiguous date (and which type of ambiguity was used).

Reply:
got it, will do another commit.